From 12405779725f9f957604bc6ac77301d70a89986e Mon Sep 17 00:00:00 2001
From: Joseph Szczypek <jszczype@redhat.com>
Date: Thu, 26 Apr 2018 13:53:28 -0400
Subject: [PATCH] [scsi] hpsa: remove old cciss-based smartarray pci ids

Message-id: <1524750809-43700-2-git-send-email-jszczype@redhat.com>
Patchwork-id: 211573
O-Subject: [RHEL8 e-stor PATCH 1/2] hpsa: remove old cciss-based smartarray pci ids
Bugzilla: 1471185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>

Per discussion with Microsemi, the old PCI IDs for smartarrays
originally used with the old cciss driver are not being
supported with RHEL8. They are also not supported with RHEL7.
Therefore the PCI IDs and the catch-all are being removed.

Acked-by: Don Brace <don.brace@microsemi.com>
Signed-off-by: Herton R. Krzesinski <herton@redhat.com>
---
 drivers/scsi/hpsa.c | 25 -------------------------
 1 file changed, 25 deletions(-)

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 3a9eca1..5a7dde4 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -81,7 +81,6 @@ MODULE_DESCRIPTION("Driver for HP Smart Array Controller version " \
 MODULE_SUPPORTED_DEVICE("HP Smart Array Controllers");
 MODULE_VERSION(HPSA_DRIVER_VERSION);
 MODULE_LICENSE("GPL");
-MODULE_ALIAS("cciss");
 
 static int hpsa_simple_mode;
 module_param(hpsa_simple_mode, int, S_IRUGO|S_IWUSR);
@@ -143,10 +142,6 @@ static const struct pci_device_id hpsa_pci_device_id[] = {
 	{PCI_VENDOR_ID_HP_3PAR, 0x0075, 0x1590, 0x007D},
 	{PCI_VENDOR_ID_HP_3PAR, 0x0075, 0x1590, 0x0088},
 	{PCI_VENDOR_ID_HP, 0x333f, 0x103c, 0x333f},
-	{PCI_VENDOR_ID_HP,     PCI_ANY_ID,	PCI_ANY_ID, PCI_ANY_ID,
-		PCI_CLASS_STORAGE_RAID << 8, 0xffff << 8, 0},
-	{PCI_VENDOR_ID_COMPAQ,     PCI_ANY_ID,	PCI_ANY_ID, PCI_ANY_ID,
-		PCI_CLASS_STORAGE_RAID << 8, 0xffff << 8, 0},
 	{0,}
 };
 
@@ -157,26 +152,6 @@ MODULE_DEVICE_TABLE(pci, hpsa_pci_device_id);
  *  access = Address of the struct of function pointers
  */
 static struct board_type products[] = {
-	{0x40700E11, "Smart Array 5300", &SA5A_access},
-	{0x40800E11, "Smart Array 5i", &SA5B_access},
-	{0x40820E11, "Smart Array 532", &SA5B_access},
-	{0x40830E11, "Smart Array 5312", &SA5B_access},
-	{0x409A0E11, "Smart Array 641", &SA5A_access},
-	{0x409B0E11, "Smart Array 642", &SA5A_access},
-	{0x409C0E11, "Smart Array 6400", &SA5A_access},
-	{0x409D0E11, "Smart Array 6400 EM", &SA5A_access},
-	{0x40910E11, "Smart Array 6i", &SA5A_access},
-	{0x3225103C, "Smart Array P600", &SA5A_access},
-	{0x3223103C, "Smart Array P800", &SA5A_access},
-	{0x3234103C, "Smart Array P400", &SA5A_access},
-	{0x3235103C, "Smart Array P400i", &SA5A_access},
-	{0x3211103C, "Smart Array E200i", &SA5A_access},
-	{0x3212103C, "Smart Array E200", &SA5A_access},
-	{0x3213103C, "Smart Array E200i", &SA5A_access},
-	{0x3214103C, "Smart Array E200i", &SA5A_access},
-	{0x3215103C, "Smart Array E200i", &SA5A_access},
-	{0x3237103C, "Smart Array E500", &SA5A_access},
-	{0x323D103C, "Smart Array P700m", &SA5A_access},
 	{0x3241103C, "Smart Array P212", &SA5_access},
 	{0x3243103C, "Smart Array P410", &SA5_access},
 	{0x3245103C, "Smart Array P410i", &SA5_access},
-- 
2.9.0

